woocommerce/woocommerce-android
# Refactor StatsRepository into stateless repository. Mark I: fetchHasOrders

Parent issue: #5090 

"Do not merge" label - we need to update FluxC's hash before merging this PR.

[Based on FluxC's PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2189)

### Description
This PR refactors fetchHasOrders into a suspendable function - removes usage of the event bus.

There are two reasons for this change

1. [The event bus architecture is considered as legacy/deprecated](https://github.com/wordpress-mobile/WordPress-FluxC-Android/wiki/%5BDeprecated%5D-Architecture)
2. WCAndroid app uses stateful repositories which need to be registered to the event bus - this leads to memory leaks when the repository doesn't unregister from the event bus. It also causes side-effects when multiple instances of a repository are created - eg. duplicate tracking events.

### Testing instructions

1. Open the dashboard
2. Verify the behaviour of fetchHasOrders has not changed
